aiaccel/apps/optimize.py

```diff
@@ -0,0 +1,154 @@
+import argparse
+import pickle as pkl
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Any
+
+from hydra.utils import instantiate
+from omegaconf import OmegaConf as oc  # noqa: N813
+
+from aiaccel.job import AbciJobExecutor
+
+"""
+Usage (if parameters are not defined in a config file):
+    python -m aiaccel.apps.optimize objective.sh params.x1="[0, 10]" params.x2="[0, 10]" --config config.yaml
+
+Usage (if parameters are defined in a config file):
+    python -m aiaccel.apps.optimize objective.sh --config config.yaml
+"""
+
+"""
+config file (yaml) example:
+
+study:
+  _target_: optuna.create_study
+  direction: minimize
+
+  sampler:
+    _target_: optuna.samplers.TPESampler
+    seed: 0
+
+params:
+  _convert_: partial
+  _target_: aiaccel.apps.optimize.HparamsManager
+  x1: [0, 1]
+  x2:
+    _target_: aiaccel.apps.optimize.SuggestFloat
+    name: x2
+    low: 0.0
+    high: 1.0
+    log: false
+
+n_trials: 30
+n_max_jobs: 4
+
+group: gaa50000
+
+"""
+
+
+@dataclass
+class Suggest:
+    def __call__(self, trial: Any) -> Any:
+        raise NotImplementedError
+
+
+@dataclass
+class Const(Suggest):
+    name: str
+    value: Any
+
+    def __call__(self, _: Any) -> Any:
+        return self.value
+
+
+@dataclass
+class SuggestFloat(Suggest):
+    name: str
+    low: float
+    high: float
+    step: float | None = None
+    log: bool | None = False
+
+    def __call__(self, trial: Any) -> float:
+        return trial.suggest_float(name=self.name, low=self.low, high=self.high, step=self.step, log=self.log)  # type: ignore
+
+
+@dataclass
+class SuggestInt(Suggest):
+    name: str
+    low: int
+    high: int
+    step: int | None = 1
+    log: bool | None = False
+
+    def __call__(self, trial: Any) -> int:
+        return trial.suggest_int(name=self.name, low=self.low, high=self.high, step=self.step, log=self.log)  # type: ignore
+
+
+@dataclass
+class SuggestXXX(Suggest): ...
+
+
+...
+
+
+class HparamsManager:
+    def __init__(self, **params_def: dict[str, int | float | list[float | int] | Suggest]) -> None:
+        self.params = {}
+        for name, param in params_def.items():
+            if not callable(param):
+                if isinstance(param, list):
+                    low, high = param
+                    param = lambda trial, name=name, low=low, high=high: SuggestFloat(name, low, high)(trial)  # type: ignore # noqa: E731
+                else:
+                    param = lambda trial, name=name, value=param: Const(name, value)(trial)  # type: ignore # noqa: E731
+            self.params[name] = param
+
+    def suggest_hparams(self, trial: Any) -> dict[str, int | float | str]:
+        return {name: param_fn(trial) for name, param_fn in self.params.items()}  # type: ignore
+
+
+def main() -> None:
+    parser = argparse.ArgumentParser()
+    parser.add_argument("job_filename", type=str, help="The shell script to execute.")
```

Comment:
こうすると，↓でPathしなくて良さそう．
```suggestion
    parser.add_argument("job_filename", type=Path, help="The shell script to execute.")
```